test/integration/quic_http_integration_test.cc

```diff
@@ -1331,6 +1331,60 @@ TEST_P(QuicHttpIntegrationTest, DeferredLogging) {
   EXPECT_EQ(/* request headers */ metrics.at(19), metrics.at(20));
 }
 
+TEST_P(QuicHttpIntegrationTest, DeferredLoggingWithBlackholedClient) {
+  config_helper_.addRuntimeOverride("envoy.reloadable_features.quic_defer_logging_to_ack_listener",
+                                    "true");
+  useAccessLog(
+      "%PROTOCOL%,%ROUNDTRIP_DURATION%,%REQUEST_DURATION%,%RESPONSE_DURATION%,%RESPONSE_"
+      "CODE%,%BYTES_RECEIVED%,%ROUTE_NAME%,%VIRTUAL_CLUSTER_NAME%,%RESPONSE_CODE_DETAILS%,%"
+      "CONNECTION_TERMINATION_DETAILS%,%START_TIME%,%UPSTREAM_HOST%,%DURATION%,%BYTES_SENT%,%"
+      "RESPONSE_FLAGS%,%DOWNSTREAM_LOCAL_ADDRESS%,%UPSTREAM_CLUSTER%,%STREAM_ID%,%DYNAMIC_"
+      "METADATA("
+      "udp.proxy.session:bytes_sent)%,%REQ(:path)%,%STREAM_INFO_REQ(:path)%");
+  initialize();
+
+  // Make a header-only request and delay the response by 1ms to ensure that the ROUNDTRIP_DURATION
+  // metric is > 0 if exists.
+  codec_client_ = makeHttpConnection(makeClientConnection(lookupPort("http")));
+  IntegrationStreamDecoderPtr response =
+      codec_client_->makeHeaderOnlyRequest(default_request_headers_);
+  absl::SleepFor(absl::Milliseconds(1));
+  waitForNextUpstreamRequest(0, TestUtility::DefaultTimeout);
+  upstream_request_->encodeHeaders(default_response_headers_, true);
+
+  // Block the main thread by not calling waitForEndStream or closing the client's connection,
+  // then wait for server to tear down connection due to too many retransmissions.
+  int iterations = 0;
+  std::string contents = TestEnvironment::readFileToStringForTest(access_log_name_);
```

Comment:
waitForAccessLog ?

Reply:
waitForAccessLog has the built-in dispatcher run-and-sleep routine for QUIC (which I added), and inlining this avoids that. I could change that waitForAccessLog to optionally skip that routine if desired.